data_safe_haven/config/backend_settings.py

```
@@ -0,0 +1,144 @@
+        if name:
+            self.logger.debug(f"Updating '[green]{name}[/]' to '{name}'.")
+            self._name = name
+        if subscription_name:
+            self.logger.debug(
+                f"Updating '[green]{subscription_name}[/]' to '{subscription_name}'."
+            )
+            self._subscription_name = subscription_name
+
+        # Write backend settings to disk (this will trigger errors for uninitialised parameters)
+        self.write()
+
+    @property
+    def admin_group_id(self) -> str:
+        if not self._admin_group_id:
+            raise DataSafeHavenParameterException(
+                "Azure administrator group not provided: use '[bright_cyan]--admin-group[/]' / '[green]-a[/]' to do so."
+            )
+        return self._admin_group_id
+
+    @property
+    def location(self) -> str:
+        if not self._location:
+            raise DataSafeHavenParameterException(
+                "Azure location not provided: use '[bright_cyan]--location[/]' / '[green]-l[/]' to do so."
+            )
+        return self._location
+
+    @property
+    def name(self) -> str:
+        if not self._name:
+            raise DataSafeHavenParameterException(
+                "Data Safe Haven deployment name not provided: use '[bright_cyan]--deployment-name[/]' / '[green]-d[/]' to do so."
+            )
+        return self._name
+
+    @property
+    def subscription_name(self) -> str:
+        if not self._subscription_name:
+            raise DataSafeHavenParameterException(
+                "Azure subscription not provided: use '[bright_cyan]--subscription[/]' / '[green]-s[/]' to do so."
+            )
+        return self._subscription_name
+
+    def read(self) -> None:
+        """Read settings from YAML file"""
+        if self.config_file_path.exists():
+            with open(self.config_file_path, "r", encoding="utf-8") as f_yaml:
+                settings = yaml.safe_load(f_yaml)
+            if isinstance(settings, dict):
```

Does `yaml.safe_load` return `None` (or similar) if it fails?
Do we want to raise an exception in that instance?

---

I think probably not (see above) but there is the possible case of a corrupted/unreadable file. Thoughts?

---

I have wrapped this in a DataSafeHavenConfigException so if it happens it will be nicely logged.